pkg/controller/deployment/deployment_controller.go

```diff
@@ -576,6 +579,73 @@ func (dc *DeploymentController) syncRollingUpdateDeployment(deployment extension
 	return dc.syncDeploymentStatus(allRSs, newRS, deployment)
 }
 
+// proportionalScaling should run only on scaling events and not during the normal rollout process.
+func (dc *DeploymentController) proportionalScaling(deployment extensions.Deployment, shouldCreate bool) error {
+	newRS, oldRSs, err := dc.getAllReplicaSets(deployment, shouldCreate)
+	if err != nil {
+		return err
+	}
+	allRSs := controller.FilterActiveReplicaSets(oldRSs)
+	maybeMaxSurge := 0
+	if deployment.Spec.Strategy.Type == extensions.RollingUpdateDeploymentStrategyType {
+		maybeMaxSurge, err = intstrutil.GetValueFromIntOrPercent(&deployment.Spec.Strategy.RollingUpdate.MaxSurge, deployment.Spec.Replicas)
+		if err != nil {
+			return err
+		}
+	}
+
+	if newRS != nil {
```

Comment:
When paused I opted for not creating a new replica set (it won't be scaled up anyway) so I need to check if the latest replica set exists.